src/main/java/org/openmrs/addonindex/rest/AddOnController.java

```diff
@@ -38,9 +41,14 @@ public AddOnController(Index index) {
 	
 	@RequestMapping(method = RequestMethod.GET, value = "/api/v1/addon")
 	public Collection<AddOnInfoSummary> search(@RequestParam(value = "type", required = false) AddOnType type,
-	                                           @RequestParam(value = "q", required = false) String query, 
-						   @RequestParam(value = "tag", required = false) String tag) throws Exception {
-		return index.search(type, query, tag);
+											   @RequestParam(value = "q", required = false) String query,
```

Comment:
In the big picture it would be nice to avoid changing method signature and adding these parameters all over the place, by just doing the query parsing entirely within the ElasticSearchIndex class.

Reply:
Yes, that's true, I have currently asked around on forums. Let's hope a java based parser that's similar to the one we're currently using shows up

Reply:
Conclusion: we didn't find a Java library for this, so we'll proceed doing it in the front end in JS.